bin/assigner/src/main.cpp

```diff
@@ -707,7 +728,7 @@ int main(int argc, char *argv[]) {
     if (std::uint8_t(gen_mode & nil::blueprint::generation_mode::ASSIGNMENTS)) {
         if (vm.count("assignment-table")) {
             assignment_table_file_name = vm["assignment-table"].as<std::string>();
-        } else {
+        } else if (!public_input_only) {
```

Comment:
if `generation_mode` includes `ASSIGNMENT` - table must be generated.
We can:
1. allow mode `ASSIGNMENTS | PUBLIC_INPUT` and generate both files
2. `ASSIGNMENTS | PUBLIC_INPUT` - impossible mode (like now), in this case here `public_input_only` - never true